python/pyarrow/_flight.pyx

```diff
@@ -536,12 +545,7 @@ cdef class FlightDescriptor(_Weakrefable):
         return self.descriptor.path
 
     def __repr__(self):
-        if self.descriptor_type == DescriptorType.PATH:
-            return "<FlightDescriptor path: {!r}>".format(self.path)
-        elif self.descriptor_type == DescriptorType.CMD:
-            return "<FlightDescriptor command: {!r}>".format(self.command)
-        else:
-            return "<FlightDescriptor type: {!r}>".format(self.descriptor_type)
+        return frombytes(self.descriptor.ToString())
```

Comment:
We also added "pyarrow." in front of the class names. For those flight specific classes, that might be less needed (in comparison to the generic "Buffer", for which it is useful to know it is a pyarrow Buffer), and would also get lengthier ("pyarrow.flight.Action ..")?

Reply:
Sorry, I've been distracted as of late. I can rework the Python parts of this so that we have custom Pythonic `__repr__` implementations. I originally did the Python changes here just because I figured I'd reduce the duplication while I was already rooting around here, but it's be better to give a good experience for Python users.